docs/book/src/reference/rust_differences.md

```diff
@@ -34,3 +34,9 @@ struct MyStruct {
     field_two: bool,
 }
 ```
+
+## Borrow Checker
+
+In Rust, the borrow checker implements Rust's [ownership system](https://doc.rust-lang.org/1.8.0/book/ownership.html)
+
+In Sway, there is no borrow checker.  This means there is no concept of ownership, borrowing, or lifetimes.  Instead, objects are copied and moved similar to C++.  Also Sway does not have any destructors nor `Drop` traits.  This means allocated memory lives for the entire transaction and is not deallocated until the end of the transaction.  A transaction can contain 65 MB of memory.
```

Comment:
I'm not sure about this one. Does the VM allocate 65 MB for every tx or is it a dynamic amount based on the program bytecode? cc @FuelLabs/client 

```suggestion
In Sway, there is no borrow checker.  This means there is no concept of ownership, borrowing, or lifetimes.  Instead, objects are copied and moved similar to C++.  Also Sway does not have any destructors nor `Drop` traits.  This means allocated memory lives for the entire transaction and is not deallocated until the end of the transaction.  A transaction may allocate up to 65 MB of memory.
```

Reply:
We have 64MB of pre-allocated memory, which we reuse for each transaction. Each transaction starts with stack and heap size 0. Each time the stack or heap is expanded, we fill expanded areas with zeros. Only initialized memory is accessible, so each memory slice will be filled with zeroes first before it can be accessed.